socorro/signature/rules.py

```diff
@@ -646,20 +669,19 @@ def action(self, crash_data, result):
         # Reduce consecutive spaces to a single space
         sig = self.CONSECUTIVE_WHITESPACE_RE.sub(' ', sig)
 
-        result['signature'] = sig
+        result.set_signature(self.name, sig)
```

Comment:
We could capture the original signature and only set it if it changed. That gives us better diagnostics about what happened.